### PR TITLE
Rework DG navigation + topbar and sidebar changes

### PR DESCRIPTION
**Pull request by pthoang**
## Description
Fixes [DG-521](https://payexjira.atlassian.net/secure/RapidBoard.jspa?rapidView=80&projectKey=DG&modal=detail&selectedIssue=DG-521)
Fixes [DG-524](https://payexjira.atlassian.net/secure/RapidBoard.jspa?rapidView=80&projectKey=DG&modal=detail&selectedIssue=DG-524)

Also fixes issue with duplicate EventListeners and handleScrollbar locking the user in certain cases, and added JacascriptDocs to Accordion

## How Has This Been Tested?
Ran all existing tests. Added Snapshot tests

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have updated the **CHANGELOG** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/App/utils/SelectPanel/index.js [hunk 1]
```diff
@@ -53,26 +59,63 @@ class NavGroup extends Component {
...
                 {/* A fully functional search box will be added later. */}
                 {/* <SearchBox routes={this.props.routes} /> */}
                 <nav className="sidebar-nav">
+                    <a href="/" className="sidebar-top">{"\n"}
+                        <img className="d-none d-lg-block" src={`${basename}img/${brand}-logo-v-small.svg`} alt="logo" />{"\n"}
+                        <div className="sidebar-version">
+                            <div>
+                                Design Guide
+                            </div>
+                            Version {process.env.version || pkg.version}
```
Comment (payex-dev) [inline src/App/utils/SelectPanel/index.js:89]: Codacy Issue found: [Unnecessary block.](https://app.codacy.com/gh/SwedbankPay/design.swedbankpay.com/pullRequest?prid=5591970)

**Comment by payex-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
- Added 1
           

Clones added
============
- src/App/Home/index.js  2
- src/App/ComponentsDocumentation/components/Topbar/index.js  1
- src/App/ComponentsDocumentation/components/Topbar/index.test.js  8
- src/App/ComponentsDocumentation/components/Sidebar/index.js  3
- src/App/routes/getting-started.test.js  4
- src/App/Core/core/Page-layout/index.js  2
- src/App/ComponentsDocumentation/Introduction/index.js  2
- src/App/routes/examples.test.js  4
- src/App/routes/components.test.js  4
- src/App/ComponentsDocumentation/components/Accordion/index.js  2
- src/App/routes/all.test.js  4
- src/App/routes/core.test.js  4
- src/App/routes/utilities.test.js  4
- src/App/Examples/index.test.js  1
         

Clones removed
==============
+ src/App/Resources/index.js  -1
+ src/scripts/main/topbar/NavMenu.js  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/SwedbankPay/design.swedbankpay.com/pullRequest?prid=5591970&bid=18321293)

**Comment by payex-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Clones added
============
- src/App/Core/core/Page-layout/index.js  2
- src/App/ComponentsDocumentation/components/Topbar/index.test.js  8
- src/App/Home/index.js  2
- src/App/Examples/index.test.js  1
- src/App/ComponentsDocumentation/components/Sidebar/index.js  3
- src/App/routes/getting-started.test.js  4
- src/App/ComponentsDocumentation/components/Accordion/index.js  2
- src/App/routes/all.test.js  4
- src/App/routes/examples.test.js  4
- src/App/routes/utilities.test.js  4
- src/App/routes/core.test.js  4
- src/App/ComponentsDocumentation/Introduction/index.js  2
- src/App/routes/components.test.js  4
- src/App/ComponentsDocumentation/components/Topbar/index.js  1
         

Clones removed
==============
+ src/App/Resources/index.js  -1
+ src/scripts/main/topbar/NavMenu.js  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/SwedbankPay/design.swedbankpay.com/pullRequest?prid=5591970&bid=18321293)

File: src/App/utils/SelectPanel/index.js (hunk 1 above)
Comment (payex-dev) [inline src/App/utils/SelectPanel/index.js:89]: Codacy Issue found: [Unnecessary block.](https://app.codacy.com/gh/SwedbankPay/design.swedbankpay.com/pullRequest?prid=5591970)

**Comment by payex-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Clones added
============
- src/App/routes/components.test.js  4
- src/App/ComponentsDocumentation/components/Accordion/index.js  2
- src/App/ComponentsDocumentation/components/Topbar/index.js  1
- src/App/Examples/index.test.js  1
- src/App/ComponentsDocumentation/Introduction/index.js  2
- src/App/routes/getting-started.test.js  4
- src/App/ComponentsDocumentation/components/Topbar/index.test.js  8
- src/App/ComponentsDocumentation/components/Sidebar/index.js  3
- src/App/Core/core/Page-layout/index.js  2
- src/App/Home/index.js  2
- src/App/routes/core.test.js  4
- src/App/routes/examples.test.js  4
- src/App/routes/all.test.js  4
- src/App/routes/utilities.test.js  4
         

Clones removed
==============
+ src/scripts/main/topbar/NavMenu.js  -1
+ src/App/Resources/index.js  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/SwedbankPay/design.swedbankpay.com/pullRequest?prid=5591970&bid=18321293)

File: src/App/utils/SelectPanel/index.js [hunk 2]
```diff
@@ -53,26 +59,63 @@ class NavGroup extends Component {
...
                 {/* A fully functional search box will be added later. */}
                 {/* <SearchBox routes={this.props.routes} /> */}
                 <nav className="sidebar-nav">
+                    <a href="/" className="sidebar-top">{"\n"}
+                        <img className="d-none d-lg-block" src={`${basename}img/${brand}-logo-v-small.svg`} alt="logo" />{"\n"}
+                        <div className="sidebar-version">
+                            <div className="sidebar-version-header">
+                                Design Guide
+                            </div>
+                            Version {process.env.version || pkg.version}
```
Comment (payex-dev) [inline src/App/utils/SelectPanel/index.js:89]: Codacy Issue found: [Unnecessary block.](https://app.codacy.com/gh/SwedbankPay/design.swedbankpay.com/pullRequest?prid=5591970)

**Comment by payex-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- src/App/routes/utilities.test.js  4
- src/App/Core/core/Page-layout/index.js  2
- src/App/Examples/index.test.js  1
- src/App/ComponentsDocumentation/components/Topbar/index.js  1
- src/App/routes/components.test.js  4
- src/App/ComponentsDocumentation/components/Topbar/index.test.js  8
- src/App/ComponentsDocumentation/Introduction/index.js  2
- src/App/Home/index.js  2
- src/App/ComponentsDocumentation/components/Sidebar/index.js  3
- src/App/routes/examples.test.js  4
- src/App/routes/core.test.js  4
- src/App/routes/getting-started.test.js  4
- src/App/routes/all.test.js  4
- src/App/ComponentsDocumentation/components/Accordion/index.js  2
         

Clones removed
==============
+ src/scripts/main/topbar/NavMenu.js  -1
+ src/App/Resources/index.js  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/SwedbankPay/design.swedbankpay.com/pullRequest?prid=5591970&bid=18321293)

**Comment by payex-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- src/App/ComponentsDocumentation/components/Sidebar/index.js  3
- src/App/Home/index.js  2
- src/App/routes/getting-started.test.js  4
- src/App/ComponentsDocumentation/components/Accordion/index.js  2
- src/App/Core/core/Page-layout/index.js  2
- src/App/routes/examples.test.js  4
- src/App/Examples/index.test.js  1
- src/App/ComponentsDocumentation/components/Topbar/index.test.js  8
- src/App/ComponentsDocumentation/components/Topbar/index.js  1
- src/App/routes/utilities.test.js  4
- src/App/routes/all.test.js  4
- src/App/ComponentsDocumentation/Introduction/index.js  2
- src/App/routes/components.test.js  4
- src/App/routes/core.test.js  4
         

Clones removed
==============
+ src/App/Resources/index.js  -1
+ src/scripts/main/topbar/NavMenu.js  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/SwedbankPay/design.swedbankpay.com/pullRequest?prid=5591970&bid=18321293)

**Comment by payex-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- src/App/ComponentsDocumentation/components/Topbar/index.test.js  8
- src/App/routes/examples.test.js  4
- src/App/ComponentsDocumentation/components/Topbar/index.js  1
- src/App/Core/core/Page-layout/index.js  2
- src/App/routes/components.test.js  4
- src/App/ComponentsDocumentation/components/Sidebar/index.js  3
- src/App/ComponentsDocumentation/components/Accordion/index.js  2
- src/App/ComponentsDocumentation/Introduction/index.js  2
- src/App/Home/index.js  2
- src/App/routes/core.test.js  4
- src/App/routes/utilities.test.js  4
- src/App/Examples/index.test.js  1
- src/App/routes/all.test.js  4
- src/App/routes/getting-started.test.js  4
         

Clones removed
==============
+ src/scripts/main/topbar/NavMenu.js  -1
+ src/App/Resources/index.js  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/SwedbankPay/design.swedbankpay.com/pullRequest?prid=5591970&bid=18321293)

**Comment by payex-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- src/App/routes/all.test.js  4
- src/App/routes/components.test.js  4
- src/App/Home/index.js  2
- src/App/routes/getting-started.test.js  4
- src/App/routes/examples.test.js  4
- src/App/Core/core/Page-layout/index.js  2
- src/App/ComponentsDocumentation/components/Topbar/index.js  1
- src/App/ComponentsDocumentation/components/Topbar/index.test.js  8
- src/App/Examples/index.test.js  1
- src/App/routes/core.test.js  4
- src/App/ComponentsDocumentation/components/Sidebar/index.js  3
- src/App/routes/utilities.test.js  4
- src/App/ComponentsDocumentation/Introduction/index.js  2
- src/App/ComponentsDocumentation/components/Accordion/index.js  2
         

Clones removed
==============
+ src/scripts/main/sidebar/index.js  -2
+ src/scripts/main/topbar/NavMenu.js  -1
+ src/App/Resources/index.js  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/SwedbankPay/design.swedbankpay.com/pullRequest?prid=5591970&bid=18321293)

File: src/App/ComponentsDocumentation/components/Toast/index.js [hunk 3]
```diff
@@ -65,7 +65,7 @@ const Options = () => (
                     <td>icon</td>
                     <td>String</td>
                     <td><CodeTags type="secondary" code={"\"\""} /></td>
-                    <td>Custom icon for the toast. See <Link to="/docs/core/icons">icons</Link> for more info.</td>
+                    <td>Custom icon for the toast. See <Link to="/core/iconogrphy">icons</Link> for more info.</td>
```
Comment (pthoang) [inline src/App/ComponentsDocumentation/components/Toast/index.js:68]: Typo

File: src/App/docutils/DocToc/index.js [hunk 4]
```diff
@@ -64,7 +64,7 @@ class DocToc extends Component {
     scrollToElement (element) {
         // window.scroll(0, element.offsetTop + 100);
```
Comment (VikingTristan) [inline src/App/docutils/DocToc/index.js:65]: Can be removed?

File: src/App/routes/docs.js [hunk 5]
```diff
@@ -2,340 +2,340 @@ module.exports = [
...
+    //             componentPath: "examples/customers"
+    //         },
+    //         {
+    //             title: "Sidebar",
+    //             path: "/docs/examples/sidebar",
+    //             componentPath: "examples/sidebar"
+    //         }
+    //     ]
+    // }
 ];
```
Comment (VikingTristan) [inline src/App/routes/docs.js:341]: File deprecated?

File: src/scripts/main/sidebar/index.js [hunk 6]
```diff
@@ -115,46 +163,46 @@ const removeActiveState = (id, group, subGroup, leaf) => {
...
-    content && content.removeEventListener("scroll", _contentScrollListener);
+    content && content.removeEventListener("scroll", sidebar._contentScrollListener);
+};
+
+const _createSidebar = sidebarQuery => {
+
+    if (_sidebars.filter(sidebar => sidebar.id === sidebarQuery.id).length > 0) {
+        const updatedSheetObject = _sidebars.filter(sidebar => sidebar.id === sidebarQuery.id)[0];
+
+        updatedSheetObject.constructSidebar(sidebarQuery);
```
Comment (pthoang) [inline src/scripts/main/sidebar/index.js:196]: Should be sidebarObject

File: src/scripts/main/topbar/NavMenu.js [hunk 7]
```diff
@@ -25,25 +35,18 @@ export default class NavMenu {
         // Find focusable elements
         this.focusedElemBeforeNav = null;
         this.focusableElements = [...this.linkContainer.querySelectorAll(FOCUSELEMENTS)];
-        this._setFirstTabStop(1); // Initially set to 1 to ignore close button during tabbing, when topbar-nav is not open
+        this._setFirstTabStop(0); // Initially set to 1 to ignore close button during tabbing, when topbar-nav is not open
```
Comment (pthoang) [inline src/scripts/main/topbar/NavMenu.js:38]: Comment should be removed

File: src/scripts/main/topbar/NavMenu.js [hunk 8]
```diff
@@ -52,31 +55,44 @@ export default class NavMenu {
...
-                    }
-
-                // TAB
-                } else if (document.activeElement === this.lastTabStop) {
-                    e.preventDefault();
-                    this.firstTabStop.focus();
-                }
-            }
-        });
+        // this.navMenuElement.addEventListener("keydown", e => {
```
Comment (pthoang) [inline src/scripts/main/topbar/NavMenu.js:76]: Should be removed

File: src/scripts/main/topbar/index.test.js [hunk 9]
```diff
@@ -132,19 +131,18 @@ describe("scripts: topbar", () => {
...
+        //     NavMenu.mock.instances[0].close.mockReturnValueOnce("true");
 
-            document.dispatchEvent(new KeyboardEvent("keydown", { key: "Escape" }));
+        //     document.dispatchEvent(new KeyboardEvent("keydown", { key: "Escape" }));
 
-            expect(NavMenu.mock.instances[0].close).toHaveBeenCalled();
-            expect(handleScrollbar).toHaveBeenCalled();
-        });
+        //     expect(NavMenu.mock.instances[0].close).toHaveBeenCalled();
+        // });
```
Comment (VikingTristan) [inline src/scripts/main/topbar/index.test.js:145]: hmm

File: src/scripts/main/topbar/index.test.js [hunk 10]
```diff
@@ -132,19 +131,18 @@ describe("scripts: topbar", () => {
             expect(NavMenu.mock.instances[0].close).toHaveBeenCalled();
         });
 
-        it("Escape closes an open navMenu and calls handleScrollbar if the current page has a vertical scrollbar", () => {
-            ReactDOM.render(<Topbar navOpen />, div);
-            document.body.classList.add("has-vscroll");
+        // it("Escape closes an open navMenu", () => {
```
Comment (pthoang) [inline src/scripts/main/topbar/index.test.js:134]: Why is this commented out?

**Comment by payex-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- src/App/routes/utilities.test.js  4
- src/App/ComponentsDocumentation/components/Topbar/index.test.js  8
- src/App/ComponentsDocumentation/components/Sidebar/index.js  3
- src/App/routes/getting-started.test.js  4
- src/App/Examples/index.test.js  1
- src/App/ComponentsDocumentation/components/Topbar/index.js  1
- src/App/docutils/ComponentPreview/index.js  2
- src/App/routes/examples.test.js  4
- src/App/Core/core/Page-layout/index.js  2
- src/App/routes/all.test.js  4
- src/App/ComponentsDocumentation/components/Accordion/index.js  2
- src/App/routes/components.test.js  4
- src/App/ComponentsDocumentation/Introduction/index.js  2
- src/App/Home/index.js  2
- src/App/routes/core.test.js  4
         

Clones removed
==============
+ src/App/Resources/index.js  -1
+ src/scripts/main/sidebar/index.js  -2
+ src/scripts/main/topbar/index.test.js  -2
+ src/scripts/main/topbar/NavMenu.js  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/SwedbankPay/design.swedbankpay.com/pullRequest?prid=5591970&bid=18321293)

**Comment by payex-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- src/App/ComponentsDocumentation/components/Sidebar/index.js  3
- src/App/routes/components.test.js  4
- src/App/Core/core/Page-layout/index.js  2
- src/App/Examples/index.test.js  1
- src/App/routes/examples.test.js  4
- src/App/routes/utilities.test.js  4
- src/App/routes/core.test.js  4
- src/App/routes/all.test.js  4
- src/App/Home/index.js  2
- src/App/routes/getting-started.test.js  4
- src/App/ComponentsDocumentation/components/Accordion/index.js  2
- src/App/docutils/ComponentPreview/index.js  2
- src/App/ComponentsDocumentation/components/Topbar/index.test.js  8
- src/App/ComponentsDocumentation/Introduction/index.js  2
- src/App/ComponentsDocumentation/components/Topbar/index.js  1
         

Clones removed
==============
+ src/scripts/main/sidebar/index.js  -2
+ src/scripts/main/topbar/index.test.js  -2
+ src/App/Resources/index.js  -1
+ src/scripts/main/topbar/NavMenu.js  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/SwedbankPay/design.swedbankpay.com/pullRequest?prid=5591970&bid=18321293)

**Pull request merged**